src/frontend/src/optimizer/plan_node/stream_group_topn.rs

```diff
@@ -43,8 +42,7 @@ impl StreamGroupTopN {
             input.functional_dependency().clone(),
             input.distribution().clone(),
             false,
-            // TODO: https://github.com/risingwavelabs/risingwave/issues/7205
-            FixedBitSet::with_capacity(logical.schema().len()),
+            logical.input().watermark_columns().clone(),
```

Comment:
maybe we can just do it on group key :thinking: The groupTopN might retract some records